source/features/unfinished-comments.tsx

```diff
@@ -10,7 +10,10 @@ function hasDraftComments(): boolean {
 	// `[id^="convert-to-issue-body"]` excludes the hidden pre-filled textareas created when opening the dropdown menu of review comments
 	return select.all('textarea:not([disabled], [id^="convert-to-issue-body"])').some(textarea =>
 		textarea.value !== textarea.textContent // Exclude comments being edited but not yet changed (and empty comment fields)
-		&& !select.exists('.btn-primary[disabled]', textarea.form!), // Exclude forms being submitted
+		&& (
+			!select.exists('.btn-primary[disabled]', textarea.form!) // Exclude forms being submitted
+			|| select.exists('input[aria-label="Title"]', textarea.form!) // Exclude forms doesn't fill in required title but have draft comments
```

Comment:
cc @cheap-glitch 

I'm stumped here. The `disabled` check was added in https://github.com/sindresorhus/refined-github/pull/3707, apparently to avoid a "click submit and immediately change tab" situation.

I tried looking into it and the same exact mechanism is used to disable the button when the form is invalid (the former via `button[data-disable-with]` and the latter with `button[data-disable-invalid]`), both affecting the same attribute. The submission does not affect other attributes.

I think the solution is to _do_ track the `submit` event (via `capture: true`, probably), and then just disable the feature for ~2 seconds after, so this filter does not have to run at all.

Something like:

```ts
function updateDocumentTitle(): void {
	if (submitting) {
		return;
	}

	// the rest
}
function disableOnSubmit() {
	clearTimeout(submitting);
	submitting = setTimeout(() => {
		submitting = undefined;
	});
}
function init(): void {
	document.addEventListener('visibilitychange', updateDocumentTitle);
	delegate(document, 'submit', disableOnSubmit)
}
```

Reply:
In short, this would replace/drop `select.exists('.btn-primary[disabled]', textarea.form)` and thus solve the issue.

Reply:
So if I understood correctly, there's no way to distinguish between an form that's invalid and one that's being submitted… :sweat: \
In that case I can't think of a better solution than to listen to the `submit` event. Should probably be a new PR though.